src/cmap/connection.ts

```diff
@@ -441,10 +449,16 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
     const message = this.prepareCommand(ns.db, command, options);
 
     let started = 0;
-    if (this.monitorCommands) {
+    if (
+      this.monitorCommands ||
+      (typeof this.description.serverConnectionId === 'bigint' &&
```

Comment:
```suggestion
      (this.description.serverConnectionId &&
```
Only need to check for existence here as we always cast it to bigint when set.

Reply:
Sounds good, didn't catch that originally!